tide-whisperer.go

```diff
@@ -38,54 +31,19 @@ type (
 		Mongo               mongo.Config        `json:"mongo"`
 		store.SchemaVersion `json:"schemaVersion"`
 	}
-
-	// so we can wrap and marshal the detailed error
-	detailedError struct {
-		Status int `json:"status"`
-		//provided to user so that we can better track down issues
-		ID              string `json:"id"`
-		Code            string `json:"code"`
-		Message         string `json:"message"`
-		InternalMessage string `json:"-"` //used only for logging so we don't want to serialize it out
-	}
-	//generic type as device data can be comprised of many things
-	deviceData map[string]interface{}
-)
-
-var (
-	errorStatusCheck       = detailedError{Status: http.StatusInternalServerError, Code: "data_status_check", Message: "checking of the status endpoint showed an error"}
-	errorNoViewPermission  = detailedError{Status: http.StatusForbidden, Code: "data_cant_view", Message: "user is not authorized to view data"}
-	errorNoPermissions     = detailedError{Status: http.StatusInternalServerError, Code: "data_perms_error", Message: "error finding permissions for user"}
-	errorRunningQuery      = detailedError{Status: http.StatusInternalServerError, Code: "data_store_error", Message: "internal server error"}
-	errorLoadingEvents     = detailedError{Status: http.StatusInternalServerError, Code: "data_marshal_error", Message: "internal server error"}
-	errorInvalidParameters = detailedError{Status: http.StatusInternalServerError, Code: "invalid_parameters", Message: "one or more parameters are invalid"}
-
-	storage store.Storage
-)
-
-const (
-	dataAPIPrefix             = "api/data "
-	medtronicLoopBoundaryDate = "2017-09-01"
-	slowQueryDuration         = 0.1 // seconds
 )
 
-//set the intenal message that we will use for logging
-func (d detailedError) setInternalMessage(internal error) detailedError {
-	d.InternalMessage = internal.Error()
-	return d
-}
-
 func main() {
 	var config Config
 
-	log.SetPrefix(dataAPIPrefix)
+	log.SetPrefix(data.DATA_API_PREFIX)
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 
 	if err := common.LoadEnvironmentConfig(
 		[]string{"TIDEPOOL_TIDE_WHISPERER_SERVICE", "TIDEPOOL_TIDE_WHISPERER_ENV"},
 		&config,
 	); err != nil {
-		log.Fatal(dataAPIPrefix, "Problem loading config: ", err)
+		log.Fatal("Problem loading config: ", err)
```

Comment:
Why do we remove the prefix ?

Reply:
I understand that setting prefix on the log at the top will display it at the beginning of the record. Adding it again in the log and it will be here twice at 2 different positions.

```go
log.SetPrefix(data.DATA_API_PREFIX)
log.Print("hello")
log.Fatal(data.DATA_API_PREFIX, "world")
```
![image](https://user-images.githubusercontent.com/33152621/78886380-62b89580-7a5e-11ea-8237-bd14fb9381cc.png)

Reply:
overall this is ugly as API is not using the same mechanism for logging (i.e. not setting an overall prefix)
maybe we should add an API logger later?